src/osd/OSD.cc

```diff
@@ -6090,7 +6091,13 @@ void OSD::handle_pg_stats_ack(MPGStatsAck *ack)
     }
   }
 
-  outstanding_pg_stats.erase(ack->get_tid());
+  // if there are earlyer pg_stats doesn't acked,
+  // this can happen when the pg_stats doesn't send
+  // successfully.
+  for (auto t : outstanding_pg_stats) {
+    if (t <= ack_tid)
```

Comment:
we cannot erase from an `std::set<>` when iterating through it. instead we should use something like:
```c++
  for (auto tid = outstanding_pg_stats.cbegin();
       tid != outstanding_pg_stats.cend(); ) {
    if (*tid <=  ack_tid) {
      tid = outstanding_pg_stats.erase(tid);
    } else {
      break;
    }
  }
```

also, since the connection does not send pg-stats out-of-order, neither does monitor ack the pg-stats out-of-order, we can assume that all tids after `ack_tid` in `outstanding_pg_stats` are greater than it. so we can `break` once `*tid >  ack_tid`.